docs/guides/configuring-custom-entities.md

```diff
@@ -0,0 +1,169 @@
+    },
+    {
+      url = {
+        type = "string",
+        required = true,
+      },
+    },
+    { created_at = typedefs.auto_timestamp_s },
+    { updated_at = typedefs.auto_timestamp_s },
+  },
+}
+```
+
+An instance of such an entity would like:
+
+```json
+{
+  "id": "385def6e-3059-4929-bb12-d205e97284c5",
+  "name": "Bobby Drop Tables",
+  "url": "https://xkcd.com/327/"
+}
+```
+
+Multiple instances of such an entity are represented as follows:
+
+```json
+{
+  "xkcds": [
+    {
+      "id": "385def6e-3059-4929-bb12-d205e97284c5",
+      "name": "bobby_tables",
+      "url": "https://xkcd.com/327/"
+    },
+    {
+      "id": "d079a632-ac8d-4a9a-860c-71de82e8fc11",
+      "name": "compiling",
+      "url": "https://xkcd.com/303/"
+    }
+  ]
+}
+```
+
+If you have more than one custom entities that you would like to configure
+then you can create other entities by specifying the entity name at the root
+level of the JSON and then list the entities as a JSON array.
+
+To configure custom entities in a DB-less instance of Kong,
+you first need to create such a JSON representation of your entities.
+
+## Configuring the custom entity secret
+
+Once you have the JSON representation, we need to store the configuration
+inside a Kubernetes Secret.
+The following command assumes the filename to be `entities.json` but you can
+use any other filename as well:
+
+```bash
+$ kubectl create secret generic -n kong kong-custom-entities --from-file=config=entities.json
+secret/kong-custom-entities created
+```
+
+Some things to note:
+- The key inside the secret should be `config`. This is not configurable at the
+  moment.
+- The secret should be accessible by the Ingress Controller. The recommended
+  practice here is to install the secret in the same namespace in which Kong
+  is running.
+ 
+## Configure the Ingress Controller
+
+Once you have the secret containing the custom entities configured,
+you need to instruct the controller to read the secret and sync the custom
+entities to Kong.
+
+To do this, you need to add the following environment variable to the
+`ingress-ccontroller` container:
+
+```yaml
+env:
+- name: CONTROLLER_KONG_CUSTOM_ENTITIES_SECRET
+  value: kong/kong-custom-entities
+```
+
+This value of the environment variable takes the form of `<namespace>/<name>`.
+You need to configure this only once.
+
+This instructs the controller to watch the above secret and configure Kong
+with any custom entities present inside the secret.
+If you change the configuration and update the secret with different entities,
+the controller will dynamically fetch the updated secret and configure Kong.
+
+## Verification
+
+You can verify that the custom entity was actually created in Kong's memory
+using the `GET /xkcds` (endpoint will differ based on the name of the entity)
+on Kong's Admin API.
+For most deployments, you will need to exec inside the Kong container and
+perform:
+```bash
+$ curl -k https://localhost:8444/<entity-name>
+```
```

Comment:
We no longer package curl with our images, and will often have no non-local listen (so spawning a test pod to run curl isn't an option:
> You can forward traffic from your local machine to the Kong Pod to access it:
```
$ kubectl port-forward kong/kong-pod-name 8444:8444
```
> and in a separate terminal:
```
 $ curl -k https://localhost:8444/<entity-name>
```